File: FE-customer/src/services/cartService.ts
import cartApi from "../api/cartApi";
import type { AddToCartRequest } from "../types/requests/cart.request";

const cartService = {
  addToCart: async (item: AddToCartRequest) => {
    const res = await cartApi.addItemToCart(item);
    return res;
  },

  getCart: async () => {
    const res = await cartApi.getCart();
    return res;
  },

  updateItemQuantity: async (itemId: string, quantity: number) => {
    const res = await cartApi.updateItemQuantity(itemId, quantity);
    return res;
  },

  removeCartItem: async (itemId: string) => {
    const res = await cartApi.removeCartItem(itemId);
    return res;
  },
};
export default cartService;

File: BE/src/main/java/com/siupo/restaurant/service/cart/CartService.java
package com.siupo.restaurant.service.cart;

import com.siupo.restaurant.dto.request.AddToCartRequest;
import com.siupo.restaurant.dto.response.CartResponse;
import com.siupo.restaurant.model.User;

public interface CartService {
    CartResponse getCartByUser(User user);
    CartResponse addItemToCart(User user, AddToCartRequest request);
    CartResponse updateItemQuantity(User user, Long itemId, Long quantity);
    CartResponse removeCartItem(User user, Long itemId);
}
